core/clustersmngr/factory_test.go

```diff
@@ -0,0 +1,118 @@
+package clustersmngr_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/cheshir/ttlcache"
+	"github.com/go-logr/logr"
+	"github.com/onsi/gomega"
+	. "github.com/onsi/gomega"
+	"github.com/weaveworks/weave-gitops/core/clustersmngr"
+	"github.com/weaveworks/weave-gitops/core/clustersmngr/fetcher"
+	"github.com/weaveworks/weave-gitops/core/nsaccess/nsaccessfakes"
+	"github.com/weaveworks/weave-gitops/pkg/server/auth"
+	"golang.org/x/net/context"
+	v1 "k8s.io/api/core/v1"
+)
+
+func TestGetImpersonatedClient(t *testing.T) {
+	g := NewGomegaWithT(t)
+	logger := logr.Discard()
+	ctx := context.Background()
+
+	ns1 := createNamespace(g)
+	ns2 := createNamespace(g)
+
+	nsChecker := &nsaccessfakes.FakeChecker{}
+	nsChecker.FilterAccessibleNamespacesReturns([]v1.Namespace{*ns2}, nil)
+
+	clustersFetcher, err := fetcher.NewSingleClusterFetcher(k8sEnv.Rest)
+	g.Expect(err).To(BeNil())
+
+	clientsFactory := clustersmngr.NewClientFactory(k8sEnv.Client, clustersFetcher, nsChecker, logger)
+	err = clientsFactory.UpdateClusters(ctx)
+	g.Expect(err).To(BeNil())
+
+	err = clientsFactory.UpdateNamespaces(ctx)
+	g.Expect(err).To(BeNil())
+
+	_, err = clientsFactory.GetImpersonatedClient(ctx, &auth.UserPrincipal{ID: "user-id"})
+	g.Expect(err).To(BeNil())
+
+	t.Run("checks all namespaces in the cluster when through the filtering", func(t *testing.T) {
+		g.Expect(nsChecker.FilterAccessibleNamespacesCallCount()).To(gomega.Equal(1))
```

Comment:
```suggestion
		g.Expect(nsChecker.FilterAccessibleNamespacesCallCount()).To(Equal(1))
```

think you have some other `gomega`s around too, not the end of the world